src/components/admin/form-components/validated-fields.jsx

```diff
@@ -0,0 +1,15 @@
+import TextField from 'material-ui/TextField';
+
+// Higher Order Components
+import withValidate from 'components/admin/form-components/withValidate';
+
+export const hasHttps = (value) => {
+  if (value && value.length > 4 && value.substr(0, 5) !== 'https') {
+    return (
+      'You are saving an image without using https.'
```

Comment:
For more generalisation, if other, non-image fields will use hasHttps as well, you could add the error message as an argument.

Reply:
But again, no need for premature optimization either, is an easy refactor if you ever want that in the future, right?

Reply:
If I were to generalize this, I would rather just change the text to "This URL requires HTTPS usage." rather than passing a new argument. I agree this may not be used for image fields, but I think validations should return errorText that describes the validation to avoid future programmer error.

Reply:
That's a good point @Helw150

Reply:
Agreed.